### PR TITLE
Refocus check whenever the window is clicked

### LINKED ISSUE #62194: Clicking near window borders may produce errors with a visible AcceptDialog in single window mode (DisplayServer regression)
### Godot version

v4.0.alpha10.official [4bbe7f0b9]

### System information

Linux Mint 20.3, Vulkan, NVIDIA GeForce GTX 1050 Ti

### Issue description

Clicking near the window border may produce the following error message. The window is no longer interactive.
```
ERROR: Condition "gui.subwindow_focused == nullptr" is true. Returning: false
   at: _sub_windows_forward_input (scene/main/viewport.cpp:2441)
```
If `borderless` is enabled, it doesn't seem to have this problem.

### Steps to reproduce

Add an `AcceptDialog`, and popup it, click near window borders.

Or, enable `interface/editor/single_window_mode`, open a dialog, click near the border.

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by Rindbee**
A refocus check is required whenever `click_on_window` is `true`.

Fix #62194.

### CODE REVIEW

**Comment by Sauermann** [comment]
When I created #68272, I did not see this PR, sorry that it did not come up during my search for related issues and PRs.

I have tested this PR and found the following usability problem:
- When I want to resize a non-focused Window, first I have to click on it so that it gets focus and afterwards I can resize it.
- I would expect, that I can resize with a single mouse-drag-and-drop-action a Window, that is not resized.

The reason for this behavior in this PR is that after
```
gui.subwindow_drag = SUB_WINDOW_DRAG_RESIZE;
```
the function
```
_sub_window_grab_focus(sw.window);
```
is called which sets
```
gui.subwindow_drag = SUB_WINDOW_DRAG_DISABLED;
```
basically disabling the resize-action immediately after enabling the resize-action.
